pkg/costmodel/costmodel.go

```diff
@@ -144,39 +144,25 @@ const (
 		label_replace(
 			label_replace(
 				avg(
-					count_over_time(kube_pod_container_resource_requests{resource="memory", unit="byte", container!="",container!="POD", node!="", %s}[%s] %s)
-					*
-					avg_over_time(kube_pod_container_resource_requests{resource="memory", unit="byte", container!="",container!="POD", node!="", %s}[%s] %s)
+					sum_over_time(kube_pod_container_resource_requests{resource="memory", unit="byte", container!="",container!="POD", node!="", %s}[%s] %s)
 				) by (namespace,container,pod,node,%s) , "container_name","$1","container","(.+)"
 			), "pod_name","$1","pod","(.+)"
 		)
 	) by (namespace,container_name,pod_name,node,%s)`
-	queryRAMUsageStr = `sort_desc(
-		avg(
-			label_replace(
-				label_replace(
-					label_replace(
-						count_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
-					), "container_name", "$1", "container", "(.+)"
-				), "pod_name", "$1", "pod", "(.+)"
-			)
-			*
+	queryRAMUsageStr = `avg(
+		label_replace(
 			label_replace(
 				label_replace(
-					label_replace(
-						avg_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
-					), "container_name", "$1", "container", "(.+)"
-				), "pod_name", "$1", "pod", "(.+)"
-			)
-		) by (namespace, container_name, pod_name, node, %s)
-	)`
+					sum_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
+				), "container_name", "$1", "container", "(.+)"
+			), "pod_name", "$1", "pod", "(.+)"
+		)
```

Comment:
@r2k1 What's the effect of changing this `count_over_time` to `sum_over_time`?

Additionally, is there any reason you chose to remove `sort_desc` from the query? Is it because the `queryCPUUsageStr` does not perform a sort?

Reply:
Performance and clarity.

`sort_desc` isn't used by any other query and have performance implications as well. There is also should be no reason to use it. I suspect it's just been left out after a manual query testing.

Reply:
I've seen many people got confused with the query in slack channel.

Reply:
@thomasvn to be clear, he's changing a (`count_over_time` * `avg_over_time`) to a `sum_over_time`. Changing a count, alone, to a sum would not make sense.

Reply:
For extra clarity here is old vs new:

## Old
```go
queryRAMUsageStr = `sort_desc(
	avg(
		label_replace(
			label_replace(
				label_replace(
					count_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
				), "container_name", "$1", "container", "(.+)"
			), "pod_name", "$1", "pod", "(.+)"
		)
		*
		label_replace(
			label_replace(
				label_replace(
					avg_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
				), "container_name", "$1", "container", "(.+)"
			), "pod_name", "$1", "pod", "(.+)"
		)
	) by (namespace, container_name, pod_name, node, %s)
)`
```

## New
```go
queryRAMUsageStr = `avg(
	label_replace(
		label_replace(
			label_replace(
				sum_over_time(container_memory_working_set_bytes{container!="", container!="POD", instance!="", %s}[%s] %s), "node", "$1", "instance", "(.+)"
			), "container_name", "$1", "container", "(.+)"
		), "pod_name", "$1", "pod", "(.+)"
	)
) by (namespace, container_name, pod_name, node, %s)`
```

Reply:
@r2k1 Agree, thanks for the explanation

Reply:
@nikovacevic Oh wow thanks! I hadn't noticed that it was a `count_over_time * avg_over_time`. That makes a lot of sense now.